app/classes/generate_new_week_assignments.rb

```diff
@@ -0,0 +1,27 @@
+class GenerateNewWeekAssignments
+  def new(week)
+    last_assigned = []
+
+    Employee.all.each do |employee|
+      last_employee_assignment = Assignment.where(employee_id: employee.id).order(:week_id).last
+
+      if last_employee_assignment.present?
+        last_assigned.push(last_employee_assignment)
+      else
+        #new employee who haven't been assigned yet dealt with here
+      end
+    end
+    
+    last_assigned = last_assigned.sort_by {|assignment| [assignment.week, assignment.employee]}
+
+    employee = last_assigned.first.employee
+
+    Assignment.create(week: week, employee: employee)
+
+    last_assigned.shift
+
+    employee = last_assigned.first.employee
+
+    Assignment.create(week: week, employee: employee)
+  end
```

Comment:
The `new` method is like a keyword method and should never be defined in a Ruby class. We use initialize instead which is called by new. The initialize method should only set up the object and we should have a call method that does the real work of finding employees and creating assignments.

Finally you should extract some of the code here into private methods in accordance with the single responsibility principle (I think this is in the first or second chapter of that Ruby book 😄). The named private methods will help me understand what your code is doing (it's hard to follow without digging into the logic).